### ISSUE TITLE
[rom_ctrl] V2(S) Signoff

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#21032]
### Description

Ensure V2(S) signoff criteria are still maintained (this is not a focus area block).

**Comment by msfschaffner** [comment]
Changes since ES tapeout include misc changes in docs, general reggen improvements. 
The only notable change in functionality was the removal of the S&P layer, which is already fully modeled and tested in DV.
No TODOs have been added to the code.

```
$ git log Earlgrey-M2.5.2-RC0..HEAD --oneline hw/ip/gpio

c721c51c13 [rtl, prim] Add 'commit' functionality to prim_count
fc8484601e [reggen,hw] Create index parameter for registers windows
de31bdf1c2 [reggen] Remove the devmode input
b6d376bc56 [dv] Fix timeout in rom_ctrl stress tests
803d9aec77 [prim, rom_ctrl] Remove S&P layer from data scrambling
f9af60328d [rom_ctrl,sival] Add a list of testable features to rom_ctrl.hjson
1b16ca2122 [reggen] Add mubi support SWAccess that sets/clears a reg
59f8142826 [doc] Moved badges over to using hosted images
2ff1f6fbae [doc] rom_ctrl registers and interfaces now use CMDGEN
d67e35d864 [reggen] Generate constants for only the main block
7688e714e8 [reggen] Add initial support for version and cip_id hjson fields
fbd888eea8 Revert "[reggen] Add CIP_IDs and bump all major versions"
0ba10b3cd3 [reggen] Add CIP_IDs and bump all major versions
```

Coverage report from 02/08/2024:



![Image](https://github.com/lowRISC/opentitan/assets/41358501/5d101480-8a51-45cf-bf52-e45e0de60544)

The currently claimed verification stage V2S is hence still fulfilled.

**Comment by msfschaffner** [comment]
@rswarbrick can you take a look at this issue and close if you agree that D2S is still fulfilled?

**Comment by rswarbrick** [comment]
I've just taken a careful look. Here are the issues from your list above that look like they will need work in this rom_ctrl before we're completely done for the block. I've added a note each time about why it shouldn't block V2S signoff.

- https://github.com/lowRISC/opentitan/issues/20887: Not needed for V2S because it's essentially about missing annotations for a lint flow.
- https://github.com/lowRISC/opentitan/issues/17501: This tracks a coverage hole. Needed for V3, but not for V2S.
- https://github.com/lowRISC/opentitan/issues/16840: Again, this is a coverage hole.


However, we've also got an item that asks for all V2/V2S tests to be written and passing for multiple seeds (90%) in nightlies. The testplan progress image that appears here doesn't quite show that: it looks like only 2/4 for V2S! That's a little pessimistic because the 2 tests that fail are only failing for some seeds. Unfortunately, the pass rates are not all above the 90% in the requirement, so I don't think we can quite sign it off.

I suspect fixing this will just be a case of fixing a silly DV bug, but I don't think we're quite ready for V2S signoff now.

I believe that the remaining work required is to get the pass rate for `rom_ctrl_passthru_mem_tl_intg_err` up from 85% (last night) to something above 90%.

**Comment by msfschaffner** [comment]
Thanks @rswarbrick, you are right. We have https://opentitan.org/book/doc/project_governance/checklist/index.html#sim_nightly_regression_v2 which mandates pass rates above 90%.

**Comment by msfschaffner** [comment]
@rswarbrick is there anything I can help with V2S signoff for this or do you have some patches in flight to get pass rates above 90%?

**Comment by rswarbrick** [comment]
I *didn't* have any in flight, but I've just pushed something that should sort things out. It turns out that VCS randomises enum expressions more strictly than Xcelium, leading to a bunch of failures in this particular test. It should be fixed with #21643.

**Comment by rswarbrick** [comment]
I spoke too soon! VCS was being completely reasonable, and now I understand a bit more about function calls in constraint solving. There's a fix that actually works in #21647.

**Comment by msfschaffner** [comment]
I reran the nightly regression with your patch and got the following statistics:
### Testplan Progress
|  Items  |  Total  |  Written  |  Passing  |  Progress  |
|:-------:|:-------:|:---------:|:---------:|:----------:|
|   V1    |    8    |     8     |     8     |  100.00 %  |
|   V2    |    6    |     6     |     6     |  100.00 %  |
|   V2S   |    4    |     4     |     4     |  100.00 %  |
|   V3    |    1    |     1     |     0     |   0.00 %   |

## Coverage Results

### [Coverage Dashboard](/home/msf/Projects/opentitan/scratch/master/rom_ctrl-sim-vcs/cov_report/dashboard.html)

|  SCORE  |  LINE   |  COND   |  TOGGLE  |   FSM    |  BRANCH  |  ASSERT  |  GROUP  |
|:-------:|:-------:|:-------:|:--------:|:--------:|:--------:|:--------:|:-------:|
| 97.43 % | 97.04 % | 92.65 % | 97.88 %  | 100.00 % | 98.37 %  | 97.89 %  | 98.14 % |

I think this looks good for signing at V2S now :).
